apps/shipments/serializers/shipment_action.py

```diff
@@ -46,3 +49,10 @@ def validate_action_type(self, action_type):
             raise exceptions.ValidationError(f'Action {action_type.name} not available while Shipment '
                                              f'is in state {TransitState(shipment.state).name}')
         return action_type
+
+    def validate_action_timestamp(self, action_timestamp):
+        if settings.PROFILES_ENABLED and not is_internal_call(self.context['request'], 'third-party-integrator'):
+            raise exceptions.ValidationError('Can only manually set timestamp for action on internal calls')
+        if action_timestamp > datetime.now(timezone.utc):
```

Comment:
Do you think there is any concern here for time drift to cause valid "current timestamp" data to be rejected?

Reply:
This could be an issue if it is reporting a "current timestamp", but so far that is not something that would occur on the third party integrator. So far, the only instances where we are, or would be, using this contain timestamps that are earlier than the current time.